File: week6/AbsEncap.cpp
#include <iostream>
using namespace std;

class AccessSpecifierDemo {
private:
    int priVar;
protected:
    int proVar;
public:
    int pubVar;

public:
    void setVar(int priVar,int proVar,int pubVar) {
        this->priVar = priVar;
        this->proVar = proVar;
        this->pubVar = pubVar;
    }
    void getVar() {
        cout << "Private: " << priVar << ", Protected: " << proVar << ", Public: " << pubVar <<endl;
    }
};

int main() {
    AccessSpecifierDemo obj;
    int a,b,c;
    cout<<"enter the values of a,b,c: "<<endl;
    cin>>a>>b>>c;
    obj.setVar(a,b,c);
    obj.getVar();
    return 0;
}

File: week6/AbsEncap.java
public class AccessSpecifierDemo{
    
    private int priVar;
    protected int proVar;
    public int pubVar;
    
    public void setVar(int priValue, int proValue, int pubValue) {
        this.priVar = priValue;
        this.proVar = proValue;
        this.pubVar = pubValue;
    }

    public void getVar() {
        System.out.println("Private Variable: " + priVar);
        System.out.println("Protected Variable: " + proVar);
        System.out.println("Public Variable: " + pubVar);
    }

 
    public static void main(String[] args) {
        AccessSpecifierDemo obj = new AccessSpecifierDemo();
        obj.setVar(11, 21, 31);
        obj.getVar();
    }
}
